helpers/helpers.go

```diff
@@ -0,0 +1,136 @@
+package helpers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cognitoidentityprovider"
+	"log/slog"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+func SetupFolders(baseDir string, logger *slog.Logger) error {
+	// create output directory and set permissions
+	err := os.Chdir(baseDir)
+	if err != nil {
+		logger.Error(err.Error())
+		os.Exit(1)
+	}
+
+	err = os.MkdirAll("output", 0777)
+	if err != nil {
+		logger.Error(err.Error())
+		os.Exit(1)
+	}
+	err = os.Chown("output", 1000, 1000)
+	if err != nil {
+		logger.Error(err.Error())
+		os.Exit(1)
+	}
+
+	err = os.MkdirAll("workspace", 0777)
+	if err != nil {
+		logger.Error(err.Error())
+		os.Exit(1)
+	}
+	err = os.Chown("workspace", 1000, 1000)
+	if err != nil {
+		logger.Error(err.Error())
+		os.Exit(1)
+	}
+
+	err = os.MkdirAll("pids", 0777)
+	if err != nil {
+		logger.Error(err.Error())
+		os.Exit(1)
+	}
+	err = os.Chown("pids", 1000, 1000)
+	if err != nil {
+		logger.Error(err.Error())
+		os.Exit(1)
+	}
+	return err
+}
+
+func GetBaseDir() string {
+	baseDir := os.Getenv("BASE_DIR")
+	if baseDir == "" {
+		baseDir = "/mnt/efs"
+	}
+	return baseDir
+}
+
+func readFile(baseDir, integrationID string) ([]byte, error) {
+	filePath := filepath.Join(baseDir, "pids", integrationID)
+	for {
+		content, err := os.ReadFile(filePath)
+		if err != nil {
+			if !os.IsNotExist(err) {
+				return nil, fmt.Errorf("error reading file: %w", err)
+			}
+		} else if len(content) > 0 {
+			// Return content if the file is not empty
+			return content, nil
+		}
+
+		time.Sleep(100 * time.Millisecond)
```

Comment:
was this for debugging? or is this something to wait indefinitely for the file to have a value?